sherpa/conftest.py

```diff
@@ -647,3 +647,8 @@ def cleanup_ds9_backend():
     #
     from sherpa.image import backend
     backend.close()
+
+@pytest.fixture(autouse=True)
+def add_sherpa_test_data_dir(doctest_namespace):
+    doctest_namespace["data_dir"] = 'sherpa-test-data/sherpatest/'
```

Comment:
Nope. We don't want to do that.

Reply:
Here's what I came up with:

```
% git diff .
diff --git a/docs/conftest.py b/docs/conftest.py
index 4bd65e31..e6b72650 100644
--- a/docs/conftest.py
+++ b/docs/conftest.py
@@ -20,6 +20,20 @@
 
 import pytest
 
+from sherpa.utils.testing import get_datadir
+
+
 @pytest.fixture(autouse=True)
 def add_sherpa_test_data_dir(doctest_namespace):
-    doctest_namespace["data_dir"] = 'sherpa-test-data/sherpatest/'
\ No newline at end of file
+
+    # We could error out here, but we only want the tests that
+    # use it to fail.
+    #
+    path = get_datadir()
+    if path is None:
+        return
+
+    if not path.endswith('/'):
+        path += '/'
+
+    doctest_namespace["data_dir"] = path
diff --git a/sherpa/conftest.py b/sherpa/conftest.py
index 9a0956d2..e7416c23 100644
--- a/sherpa/conftest.py
+++ b/sherpa/conftest.py
@@ -648,7 +648,18 @@ def cleanup_ds9_backend():
     from sherpa.image import backend
     backend.close()
 
+
 @pytest.fixture(autouse=True)
 def add_sherpa_test_data_dir(doctest_namespace):
-    doctest_namespace["data_dir"] = 'sherpa-test-data/sherpatest/'
+
+    # We do not want to use test_data_path/make_data_path as this
+    # errors out when the data is not available.
+    #
+    path = get_datadir()
+    if path is not None:
+        if not path.endswith('/'):
+            path += '/'
+
+        doctest_namespace["data_dir"] = path
+
     doctest_namespace["data_3c273"] = 'sherpa/astro/datastack/tests/data/'
```

The doc tests that require the data fails now fail with

```
068   >>> data.set_arf(read_arf(data_dir + '9774.arf'))
UNEXPECTED EXCEPTION: NameError("name 'data_dir' is not defined")
Traceback (most recent call last):
  File "/home/dburke/miniconda3/envs/sherpa-work/lib/python3.10/doctest.py", line 1350, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest simulate.rst[17]>", line 1, in <module>
NameError: name 'data_dir' is not defined
/home/dburke/sherpa/sherpa-work/docs/evaluation/simulate.rst:68: UnexpectedException
```

I thought about setting the path to something like "data-does-not-exist/", so the error message would come from the read call (and it may catch cases where we have actually defined `data_dir` somewhere), but this looks cleaner to identify.

It doesn't look like there's a nice way to mark these tests as "skip if data directory is not installed".

Reply:
> It doesn't look like there's a nice way to mark these tests as "skip if data directory is not installed".

That's because in a narrative text, there is no easy telling where one example begins and the other one ends (recall that variables are kept form the beginning to the end of the file so you can interleave code and explanatory text). So, the only option is really to step testing the rest of the file at that point.

And let's not be more picky than we have to be. We are not going to run doctests that same way we do other CI on different operating systems, with different version of numpy and Python, etc. I envision that we'll add doctests to just one single runner eventually, and for now it's only run locally be developers you want to make sure the docs still work or their new examples are complete. So, I'm going with "if the test data is not available, raise an exception and stop altogether". I don't think we have to try too hard to work around that.